crates/storage/provider/src/transaction.rs

```diff
@@ -489,30 +490,19 @@ where
 
         self.insert_hashes(first_number..=last_block_number, last_block_hash, expected_state_root)?;
 
+        self.calculate_history_indices(first_number..=last_block_number)?;
+
         // Update pipeline progress
         self.update_pipeline_stages(new_tip_number)?;
 
         Ok(())
     }
 
-    /// Insert full block and make it canonical.
-    ///
-    /// This inserts the block and builds history related indexes. Once all blocks in a chain have
-    /// been committed, the state root needs to be inserted separately with
-    /// [`Transaction::insert_hashes`].
-    ///
-    /// # Note
-    ///
-    /// This assumes that we are using beacon consensus and that the block is post-merge, which
-    /// means that the block will have no block reward.
-    /// TODO do multi block insertion.
-    pub fn insert_block(&mut self, block: SealedBlockWithSenders) -> Result<(), TransactionError> {
```

Comment:
this function was public, 
is this no longer used? because all we need is `insert_canonical_block` and `calculate_history_indices` and removing the function prevents us from using this incorrectly?

Reply:
I think there was an idea to have a splited (more granular) function for potential use elsewhere, but yeah, it is used only in one place and in one test.

I returned it back and sorted out the description as the scope of function was reduced to only inserting block.